kep/108-Memory-bandwidth-and-LLC-aware-filtering-and-scoring/README.md

```diff
@@ -74,42 +74,44 @@ the detail about filtering and scoring will be described below.
+
+For instance, there is a workload, it can be classified as a "stream" workload, according to our off-line analysis this kind of workload consume 5 GB/s memory bandwidth and 0.2 GB memory in average (that's to say its MEMORY_BANDWIDTH_PROFILE is 5 GB/s, its MEMORY_UTILIZATION_PROFILE is 0.2 GB). Consider OVERPROVISIONING factor, the memory bandwidth and memory requirement is 5 * 2 = 10 (GB/s) and 0.2 * 2 = 0.4 (GB). The system free memory bandwidth and memory should be larger than the requirement. 
 
 . Scoring
 
 For every pod each node will be assigned a list of scores based on:
 
 * available resources when sorted by the scoring metrics (see above)
-    * This means that for a pod, all the nodes will be sorted by memory bandwidth, memory latency, cpu, LLC contention and LLC utilization
+    * This means that for a pod, all the nodes will be sorted by memory bandwidth, memory latency, CPU, LLC contention and LLC utilization
     * All these lists will then go on to the next step
 
 
 * the number of counterweights that are applicable to the node
-    * A counterweight is a pod that has already been deployed on a node, but we assume it is still too soon to notice its impact in the measurements we receive from Prometheus. Therefore, we add its profile (memory bandwidth, cpu, and LLC utilization for now) as a counterweight for a *COUNTERWEIGTH_EXPIRATION_POLICY* period of time to the respective node
+    * A counterweight is the resources reserved for a pod that has already been scheduled to a node. We assume it is still too soon to notice its impact in the measurements we receive from Prometheus. Therefore, we add its profile (memory bandwidth, CPU, and LLC utilization for now) as a counterweight for a *COUNTERWEIGTH_EXPIRATION_POLICY* period of time to the respective node. It is to avoid scenarios where a bulk of workloads all get scheduled on the same node because they don't start using resources.
 
 * ranking strategy when giving point to the sorted nodes
     * We allow flexible point awarding for our scheduler plugin. For example, our default strategy is to award 10 points to the node ranking first in a certain resource, 5 points to the second place, 1 point to the 3rd place and 0 points to anyone else.
 
 * the affinity to the respective resource
-    * The affinity is also obtained via a top-down analysis. It can have values between 0 (no impact) and 100 (very heavy impact) of a resource upon the pod execution time. The affinity of a resource acts a multiplier for the points awared in the ranking algorithm.
-       Therefore, a node with a lot of available memory bandwith (ranking first) might not get a pod that needs less LLC thrashing in fact.
+    * The affinity is also obtained via a top-down analysis. It can have values between 0 (no impact) and 100 (very heavy impact) of a resource upon the pod execution time. The affinity of a resource acts a multiplier for the points awarded in the ranking algorithm.
```

Comment:
The term "top-down analysis" is mentioned multiple times in this KEP. Are there any design/implementation docs to explain how this works in detail?

Reply:
There isn't a doc. For instance, image recognition. there are a lot of such kind of workload, we test a serials of different image recognition workloads and summarize the average resource usage and the resource impaction.  The average usage values is "profile". the impaction is "affinity", it is normalized to 0 - 100.